### PR TITLE
Remove the erroring on force-dynamic in static generation for app route

### PR DESCRIPTION
**Pull request by huozhi**
### What
* Remove the erroring of force-dynamic is not able to use during static generation
* Export the route segments properly in sitemap conventions

### Why

We discovered this error is showing up when users are using force-dynamic with generating multi sitemaps.

When you have a dynamic `route /[id]/route.js` , and you have generateSitemaps defined which is actually using `generateSitemaps` under the hood , then you set the route to dynamic with `export dynamic = 'force-dynamic'`.

We should keep the route still as dynamic. `generateStaticParams` is only for generating the paths, which is static in build time. And the `force-dynamic` is going to be applied to each generated path.

Closes NEXT-2881

### CODE REVIEW

**Comment by ijjk** [comment]
# Stats from current PR

<details open>
<summary><strong>Default Build</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | vercel/next.js canary  | vercel/next.js fix/sitemap-force-dynamic | Change |
| - | - | - | - |
| buildDuration | 14s | 14s | N/A |
| buildDurationCached | 7.4s | 6.2s | N/A |
| nodeModulesSize | 198 MB | 198 MB | N/A |
| nextStartRea..uration (ms) | 438ms | 438ms | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack)</strong></summary>

|  | vercel/next.js canary  | vercel/next.js fix/sitemap-force-dynamic | Change |
| - | - | - | - |
| 2453-HASH.js gzip | 30.8 kB | 30.8 kB | N/A |
| 3304.HASH.js gzip | 181 B | 181 B | ✓ |
| 3f784ff6-HASH.js gzip | 53.7 kB | 53.7 kB | ✓ |
| 8299-HASH.js gzip | 5.04 kB | 5.04 kB | N/A |
| framework-HASH.js gzip | 45.2 kB | 45.2 kB | ✓ |
| main-app-HASH.js gzip | 242 B | 242 B | ✓ |
| main-HASH.js gzip | 32.2 kB | 32.2 kB | N/A |
| webpack-HASH.js gzip | 1.68 kB | 1.68 kB | N/A |
| Overall change | 99.3 kB | 99.3 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | vercel/next.js fix/sitemap-force-dynamic | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | vercel/next.js fix/sitemap-force-dynamic | Change |
| - | - | - | - |
| _app-HASH.js gzip | 196 B | 197 B | N/A |
| _error-HASH.js gzip | 184 B | 184 B | ✓ |
| amp-HASH.js gzip | 505 B | 505 B | ✓ |
| css-HASH.js gzip | 324 B | 325 B | N/A |
| dynamic-HASH.js gzip | 2.5 kB | 2.5 kB | N/A |
| edge-ssr-HASH.js gzip | 258 B | 258 B | ✓ |
| head-HASH.js gzip | 352 B | 352 B | ✓ |
| hooks-HASH.js gzip | 370 B | 371 B | N/A |
| image-HASH.js gzip | 4.21 kB | 4.21 kB | ✓ |
| index-HASH.js gzip | 259 B | 259 B | ✓ |
| link-HASH.js gzip | 2.67 kB | 2.67 kB | N/A |
| routerDirect..HASH.js gzip | 314 B | 312 B | N/A |
| script-HASH.js gzip | 386 B | 386 B | ✓ |
| withRouter-HASH.js gzip | 309 B | 309 B | ✓ |
| 1afbb74e6ecf..834.css gzip | 106 B | 106 B | ✓ |
| Overall change | 6.57 kB | 6.57 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | vercel/next.js fix/sitemap-force-dynamic | Change |
| - | - | - | - |
| _buildManifest.js gzip | 481 B | 484 B | N/A |
| Overall change | 0 B | 0 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | vercel/next.js fix/sitemap-force-dynamic | Change |
| - | - | - | - |
| index.html gzip | 529 B | 528 B | N/A |
| link.html gzip | 541 B | 540 B | N/A |
| withRouter.html gzip | 524 B | 522 B | N/A |
| Overall change | 0 B | 0 B | ✓ |

</details>

<details>
<summary><strong>Edge SSR bundle Size</strong></summary>

|  | vercel/next.js canary  | vercel/next.js fix/sitemap-force-dynamic | Change |
| - | - | - | - |
| edge-ssr.js gzip | 95.3 kB | 95.3 kB | N/A |
| page.js gzip | 3.04 kB | 3.04 kB | N/A |
| Overall change | 0 B | 0 B | ✓ |

</details>

<details>
<summary><strong>Middleware size</strong></summary>

|  | vercel/next.js canary  | vercel/next.js fix/sitemap-force-dynamic | Change |
| - | - | - | - |
| middleware-b..fest.js gzip | 625 B | 625 B | ✓ |
| middleware-r..fest.js gzip | 151 B | 151 B | ✓ |
| middleware.js gzip | 25.5 kB | 25.5 kB | N/A |
| edge-runtime..pack.js gzip | 839 B | 839 B | ✓ |
| Overall change | 1.61 kB | 1.61 kB | ✓ |

</details>

<details>
<summary><strong>Next Runtimes</strong></summary>

|  | vercel/next.js canary  | vercel/next.js fix/sitemap-force-dynamic | Change |
| - | - | - | - |
| app-page-exp...dev.js gzip | 170 kB | 170 kB | ✓ |
| app-page-exp..prod.js gzip | 96.9 kB | 96.9 kB | ✓ |
| app-page-tur..prod.js gzip | 98.6 kB | 98.6 kB | ✓ |
| app-page-tur..prod.js gzip | 93.1 kB | 93.1 kB | ✓ |
| app-page.run...dev.js gzip | 144 kB | 144 kB | ✓ |
| app-page.run..prod.js gzip | 91.5 kB | 91.5 kB | ✓ |
| app-route-ex...dev.js gzip | 21.4 kB | 21.3 kB | N/A |
| app-route-ex..prod.js gzip | 15.1 kB | 15.1 kB | N/A |
| app-route-tu..prod.js gzip | 15.1 kB | 15.1 kB | N/A |
| app-route-tu..prod.js gzip | 14.8 kB | 14.8 kB | N/A |
| app-route.ru...dev.js gzip | 21 kB | 21 kB | N/A |
| app-route.ru..prod.js gzip | 14.8 kB | 14.8 kB | N/A |
| pages-api-tu..prod.js gzip | 9.54 kB | 9.54 kB | ✓ |
| pages-api.ru...dev.js gzip | 9.81 kB | 9.81 kB | ✓ |
| pages-api.ru..prod.js gzip | 9.54 kB | 9.54 kB | ✓ |
| pages-turbo...prod.js gzip | 22.4 kB | 22.4 kB | ✓ |
| pages.runtim...dev.js gzip | 23.1 kB | 23.1 kB | ✓ |
| pages.runtim..prod.js gzip | 22.4 kB | 22.4 kB | ✓ |
| server.runti..prod.js gzip | 50.9 kB | 50.9 kB | ✓ |
| Overall change | 842 kB | 842 kB | ✓ |

</details>

<details>
<summary><strong>build cache</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | vercel/next.js fix/sitemap-force-dynamic | Change |
| - | - | - | - |
| 0.pack gzip | 1.57 MB | 1.56 MB | N/A |
| index.pack gzip | 106 kB | 106 kB | ⚠️ +235 B |
| Overall change | 106 kB | 106 kB | ⚠️ +235 B |

</details>

<details>
<summary><strong>Diff details</strong></summary>

<details>
<summary>Diff for <strong>middleware.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>app-route-ex..ntime.dev.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>app-route-ex..time.prod.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>app-route-tu..time.prod.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>app-route-tu..time.prod.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>app-route.runtime.dev.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>app-route.ru..time.prod.js</strong></summary>

Diff too large to display
</details>

</details>

</details>
Commit: 306e024034b3ef2469fac9d4aec27378d000dc14

File: test/development/app-dir/dynamic-in-generate-params/index.test.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
+import { type NextInstance, createNextDescribe } from 'e2e-utils'
+
+async function getLastModifiedTime(next: NextInstance, pathname: string) {
+  const content = await (await next.fetch(pathname)).text()
+  return content.match(/<lastmod>([^<]+)<\/lastmod>/)[1]
+}
+
+createNextDescribe(
+  'app-dir - dynamic in generate params',
```
Comment (ijjk) [inline test/development/app-dir/dynamic-in-generate-params/index.test.ts:9]: Why do we want this as specifically `development` test, would it be good to have as a normal `e2e` test so we ensure both dev and start are working as expected?

Comment (huozhi) [reply test/development/app-dir/dynamic-in-generate-params/index.test.ts]: def an accident 😂 was intended to put them in e2e..

**Comment by ijjk** [comment]
## Failing test suites 
Commit: f245d2a9260edb406eacdff4341891433d915dee

`pnpm test-start test/e2e/middleware-rewrites/test/index.test.ts`
- Middleware Rewrite > should handle catch-all rewrite correctly

<details>
<summary>Expand output</summary>

  ● Middleware Rewrite › should handle catch-all rewrite correctly

    expect(received).toBe(expected) // Object.is equality

    Expected: true
    Received: false

      38 |
      39 |         await retry(async () => {
    > 40 |           expect(requests.some((item) => item.includes('article'))).toBe(true)
         |                                                                     ^
      41 |         })
      42 |       }
      43 |

      at toBe (e2e/middleware-rewrites/test/index.test.ts:40:69)
      at fn (lib/next-test-utils.ts:724:20)
      at Object.<anonymous> (e2e/middleware-rewrites/test/index.test.ts:39:9)

</details>


Read more about building and testing Next.js in [contributing.md](https://github.com/vercel/next.js/blob/canary/contributing.md).

**Review by ijjk — APPROVED** [review]

**Pull request merged**